src/pages/lessons/[slug].tsx

```diff
@@ -317,13 +316,19 @@ const Lesson: FunctionComponent<LessonProps> = ({initialLesson}) => {
         }
         break
       case 'completed':
-        console.debug('handling a change to completed', {lesson})
+        let completed = false
+        console.debug('handling a change to completed', {lesson, lessonView})
         onEnded(lesson).then((lessonView: any) => {
           if (lessonView) {
             setLessonView(lessonView)
           }
-          completeVideo(lessonView)
+          if (!completed) completeVideo(lessonView)
         })
+
+        if (lessonView) {
```

Comment:
if we have a `lessonView` in state, we can "complete" instead of waiting for the async saving of the lesson view. The makes the wait a lot less at the end of a lesson, but will also capture a lesson view if needed (if the user skipped to the end before we were able to get the lesson view, for instance)